pyproject.toml

```diff
@@ -54,6 +54,8 @@ dev = [
     "pytest-mock==3.14.0",
     "pytest-rerunfailures==14.0",
     "pytest-time>=0.3.1",
+    # Pin requests because of https://github.com/msabramo/requests-unixsocket/issues/73
```

Comment:
Which upstream bug is more correct, that one or this one? https://github.com/psf/requests/issues/6707

Reply:
as far as I can tell this is not a requests bug but an apparently breaking change that will stay as is:
![image](https://github.com/canonical/craft-application/assets/1951299/306f858a-da4e-4652-b105-7b52e5b6ed79)
so if my read of all this is correct, this needs to be fixed in requests-unixsocket

Reply:
Makes sense!